### PR TITLE
Refactored ansible-mongo script

### PR DESCRIPTION
**Pull request by pkandathil**
Removed unused packages:
- fluentd

Added kitchen test to verify if the mongo_server and client are installed.
Added ability to control the version of mongo that is installed.

### CODE REVIEW

File: .kitchen.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  hosts: test-kitchen
+  # list ansible galaxy requirements from meta/main.yml
+  requirements_path: requirements.yml
+
+platforms:
+  - name: ubuntu-14.04
+
+suites:
+  # suites (tests) found at /test/integration/$test-name . By default it will look for $test-name.yml
+  - name: default
```
Comment (killerwails) [inline .kitchen.yml]: New line plz

File: .kitchen/docker_id_rsa.pub [hunk 2]
```diff
@@ -0,0 +1 @@
+ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC/LERS5fcLLhWUHJ4wbDenSmeanqLToWRiQmiV0pgTY9qhiD/hwTlt/fpZ/Qd/xsrFNhdDHbfCC4hbp6M7Sds4Fyb/8BtivjJn5ASw01rLPArK3BvXOePUjg278iq4l5fs0/iNPUrA+QUhHpS2YSumHjEra9oXDaX3jql5/eXZ3Y2KZPDBQs3q/MAaHnzB//H+QCgs/iE0TIQRfDKjlNsmIcjgodA0OcNAV2qog7I9eU1jUFkFc2XTNJrZ5bsYq0bWhxPN52mfbJ9+X4R/erxgawYV67a+kDSSKUKVqam14reLpgqP8wuQ9xBaY5s6kgXVSB57CNTQOEwmvgApZdOP kitchen_docker_key
```
Comment (killerwails) [inline .kitchen/docker_id_rsa.pub]: Why are these keys needed?  Are they not generated every time?

File: .kitchen/logs/kitchen.log [hunk 3]
```diff
@@ -0,0 +1,10 @@
+I, [2016-03-21T11:30:29.958346 #24565]  INFO -- Kitchen: -----> Starting Kitchen (v1.6.0)
```
Comment (killerwails) [inline .kitchen/logs/kitchen.log]: Please do not include this folder

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+script:
+  # run kitchen tests (destroy, create, converge, setup, verify and destroy)
+  - bundle exec kitchen test
+
+notifications:
+  slack:
+    on_success: change
+    on_failure: always
+    rooms:
+      - secure: MwE/AEu48AT3Wm15rfQmoZCDW1ZP/72uMqbyO86BBSbc650iIeGF6oU1nm7NNo8YWfadRoSRKxmLf64hpS5OAEUL4aoYt2mElDO+p5fdYzNwx7xaI+Lr8+YVOBfrPY19I101L0/p6NgoQZko4uFKtT3Jrzkm8FJQHyQtJdclHhMKOfH+MYR7fhJ+n8BFBnoiT6PT1M3zfleo6LfDvLxj3S9cgAYxA2q37OiMOPF+0NrAIU8/TLg1RomH225S5AZp04zT8B2+xa/LY5w9gVbEReH9Lk/CobiQ5Vj+PFbSuKaxiCV/s57jypZvKIVxKIpzYUWliUMLo6Ym12Z0yvqVZVMp0ovxS50mz9nYneowcX7c0UcY1eLBvXgmVnpvvBW4RgP6nnVViQ7WtYlV4aKdZpsi/4zVNH71R/ZcLndwZ6KUqk8wMn8eiPC++WhxdJ+zYimZLiTIP5dwY5dCl7F1F/TNSEWSFzxvtCHVl6E+9W7XmR7sxLwQRX2nrxqlquabhsM58HkNLZMfIoG9/0WQ815FuhsEQNAEIag0ockpZX7IdWOEt0sTyClpdX7be+xlXzgNgd6YFdMwYgzel44YvABOwt4TeVzCoPEVdw58GQsO+xe19/kEHIrPIHUULAyxJBZLCB4VxLyKmPvXgylYvYR61uNXRWClu9Wc/h7htKE=
```
Comment (killerwails) [inline .travis.yml]: New line plz

File: defaults/main.yml [hunk 5]
```diff
@@ -11,3 +11,4 @@ mongodb_client: no
 mongodb_server: no
 mongodb_accepts_external_connections: yes
 mongodb_auth_enabled: no
+mongodb_set_permissions: yes
```
Comment (killerwails) [inline defaults/main.yml]: New line plz

File: meta/main.yml [hunk 6]
```diff
@@ -5,13 +5,7 @@ dependencies:
     repository_url: "deb http://downloads-distro.mongodb.org/repo/ubuntu-upstart dist 10gen"
   - role: telusdigital.logrotate
     logrotate_name: mongod
-  - role: telusdigital.fluentd
-    fluentd_sources:
-      - path: "{{ mongodb_log_path }}"
-        format: '^(?<date_time>(?<date>[^T]*)T(?<time>[^\+]*)\+(?<time_offset>\d{4})) \[(?<placeholder_name>[^]]*)\] (?<message>.*)'
-        tag: mongodb.service
 galaxy_info:
-  author: "Chris Olstrom"
```
Comment (killerwails) [inline meta/main.yml]: Why did you remove this line?

File: requirements.yml [hunk 7]
```diff
@@ -0,0 +1,3 @@
+---
+- src: telusdigital.apt-repository
```
Comment (killerwails) [inline requirements.yml]: New. Line. Plz

File: tasks/install.yml [hunk 8]
```diff
@@ -7,12 +7,6 @@
     - mongodb-org-server
   when: mongodb_server
 
-- name: Install Packages | pip
-  pip:
-    state: latest
-    name: pymongo
-  when: mongodb_server
-
```
Comment (killerwails) [inline tasks/install.yml]: Why is this package no longer installed?

File: meta/main.yml (hunk 6 above)
Comment (pkandathil) [inline meta/main.yml]: Since we are all going to be authors now. It's no longer relevant since he is not working at the company.

File: tasks/install.yml (hunk 8 above)
Comment (pkandathil) [inline tasks/install.yml]: We don't use the python mongo package. Not sure why it was installed in the first place.

Comment (killerwails) [inline tasks/install.yml]: We as in for your project?  Or is it not used in any other project?

File: .kitchen/docker_id_rsa.pub (hunk 2 above)
Comment (pkandathil) [inline .kitchen/docker_id_rsa.pub]: They are not generated if they exist. Hence I checked it in. We have the same files checked into this repo by Ben. I followed the same format. 
https://github.com/telusdigital/ansible-log-courier/tree/master/.kitchen

File: meta/main.yml (hunk 6 above)
Comment (aaronpederson) [inline meta/main.yml]: These roles don't belong to the company - anybody submitting changes should add themselves to the contributors section in the readme

File: tasks/install.yml (hunk 8 above)
Comment (colstrom) [inline tasks/install.yml]: It's there because *Ansible* needs it. This is the package that Ansible uses to interface with and configure mongo.

File: meta/main.yml (hunk 6 above)
Comment (colstrom) [inline meta/main.yml]: Even if we put that fact aside for a moment, Ansible Galaxy considers `author` to be a required field, and fails to import roles without it.

Comment (pkandathil) [inline meta/main.yml]: Ah ok, do we have TELUS Digital author or do I leave it is as 'Chris Olstrom'?

File: tasks/install.yml (hunk 8 above)
Comment (pkandathil) [inline tasks/install.yml]: Sweet I will add it back in. Thanks for that.

File: tasks/install.yml [hunk 9]
```diff
@@ -1,13 +1,29 @@
...
+- name: Configure Mongo | Create a list file for MongoDB.
+  apt_repository: 
+    repo: "{{ mongodb_list_file }}"
+    state: present
+
 - name: Install Packages | apt
-  apt:
-    state: latest
+  apt: 
     pkg: "{{ item }}"
```
Comment (aaronpederson) [inline tasks/install.yml]: `pkg` should be `name` as per 2.0 conventions

File: test/integration/default/serverspec/localhost/default_spec.rb [hunk 10]
```diff
@@ -1,15 +1,10 @@
 require 'spec_helper'
 
 describe 'ansible-mongodb::default' do
-
   describe package('mongodb-org-server') do
     it { should be_installed.by('apt') }
   end
```
Comment (aaronpederson) [inline test/integration/default/serverspec/localhost/default_spec.rb]: You'll need to keep this test if the pymongo role is still present

Comment (pkandathil) [inline test/integration/default/serverspec/localhost/default_spec.rb]: So the test doesn't pass even though pymongo is installed via pip. 
` 1) ansible-mongodb::default Package "pymongo" should be installed by "pip" with version "3.2.2"
            Failure/Error: it { should be_installed.by('pip').with_version('3.2.2')}
              expected Package "pymongo" to be installed by "pip" with version "3.2.2"
              /bin/sh -c pip\ list\ \|\ grep\ -iw\ --\ \\\^pymongo\ \|\ grep\ -w\ --\ 3.2.2
              
            # /tmp/verifier/suites/serverspec/localhost/default_spec.rb:13:in `block (3 levels) in <top (required)>'
`
Have you implemented this test successfully elsewhere?

**Comment by killerwails** [comment]
team approved
